src/Catalog/Helpers/Utils.cs

```diff
@@ -32,7 +34,15 @@ public static Stream GetResourceStream(string resName)
 
         public static string GetResource(string resName)
         {
-            return new StreamReader(GetResourceStream(resName)).ReadToEnd();
+            if (!_resourceStringCache.ContainsKey(resName))
+            {
+                using (var reader = new StreamReader(GetResourceStream(resName)))
+                {
+                    _resourceStringCache[resName] = reader.ReadToEnd();
```

Comment:
The embedded resources never change (unless on redeploy, which will also clear this cache)